app/models/room.rb

```diff
@@ -82,12 +92,28 @@ def set_meeting_id
 
   # Autocreate all meeting options using the default values
   def create_meeting_options
-    MeetingOption.all.find_each do |option|
-      RoomMeetingOption.create(room: self, meeting_option: option, value: option.default_value)
+    access_codes_names = %w[glViewerAccessCode ModeratorAccessCode]
+    access_configs = MeetingOption.access_codes_configs(provider: 'greenlight').to_h
+
+    MeetingOption.find_each do |option|
+      value = if access_codes_names.include?(option.name) && access_configs[option.name] == 'true'
+                generate_code
+              else
+                option.default_value
+              end
+
+      RoomMeetingOption.create(room: self, meeting_option: option, value:)
     end
   end
 
   def generate_code
     SecureRandom.alphanumeric(6).downcase
   end
+
+  def auto_generate_access_codes
+    access_configs = MeetingOption.access_codes_configs(provider: 'greenlight').to_h
+
+    generate_viewer_access_code if access_configs['glViewerAccessCode'] == 'true' && viewer_access_code.blank?
```

Comment:
The order ensures that access codes records will be lazy loaded,
A possible optimization is to invert the flow of this logic eager loading the access codes (because they are more likely to be accessed) and only lazy load the configs when needed (if the access code is `:blank?` then we check if the config value is force enabled).

I've chosen this flow because it's a bit simpler but the other still worth mentioning for future refactoring targeting optimization,